### PR TITLE
Updated teleop-frontend, fixed a bug with the colours in the map, and refactored Parameters.cpp

### PR DESCRIPTION
**Pull request by philippewarren**
… refactored Parameters.cpp constructor for more uniformity

### CODE REVIEW

File: map_image_generator/src/drawers/OccupancyGridImageDrawer.cpp [hunk 1]
```diff
@@ -80,7 +80,7 @@ void OccupancyGridImageDrawer::drawNotScaledOccupancyGridImage()
         for (auto x = 0; x < m_notScaledOccupancyGridImage.cols; x++)
         {
             int dataIndex = y * m_notScaledOccupancyGridImage.cols + x;
-            int dataValue = static_cast<uchar>(m_lastOccupancyGrid->data[dataIndex]);
+            int8_t dataValue = m_lastOccupancyGrid->data[dataIndex];
```
Comment (philippewarren) [inline map_image_generator/src/drawers/OccupancyGridImageDrawer.cpp:83]: Fix the map colour bug, part 1

File: map_image_generator/src/Parameters.cpp [hunk 2]
```diff
@@ -3,62 +3,106 @@
...
+    }
+
+    template <>
+    cv::Vec3b parseColour<cv::Vec3b>(const std::string& colour)
+    {
+        int r, g, b;
+        stringstream ss(colour);
+        ss >> r >> g >> b;
+
+        return {static_cast<uchar>(b), static_cast<uchar>(g), static_cast<uchar>(r)};
```
Comment (philippewarren) [inline map_image_generator/src/Parameters.cpp:35]: Changing back the types to `int` and added static casts, fixes the map colour bug, part 1

File: map_image_generator/src/Parameters.cpp [hunk 3]
```diff
@@ -3,62 +3,106 @@
...
+    // Defined only if T is cv::Scalar or cv::Vec3b
+    template <typename T>
+    std::enable_if_t<
+        std::is_same<T, cv::Scalar>::value || std::is_same<T, cv::Vec3b>::value, T>
+    getParam(const std::string& name, const std::string& defaultValue)
+    {
+        return makeColour<T>(name, defaultValue);
+    }
+}
+
```
Comment (philippewarren) [inline map_image_generator/src/Parameters.cpp:69]: Non-trivial use of templates, but makes the constructor par uniform and way nicer.
Is the tradeoff worth it? I think it is.

File: map_image_generator/src/Parameters.cpp [hunk 4]
```diff
@@ -3,62 +3,106 @@
...
+    m_goalColor = getParam<cv::Scalar>("goal_color", "0 175 0 255");
+    m_laserScanColor = getParam<cv::Scalar>("laser_scan_color", "255 0 0 255");
+    m_textColor = getParam<cv::Scalar>("text_color", "255 255 255 255");
+
+    m_globalPathThickness = getParam<int>("global_path_thickness", 3);
+    m_robotSize = getParam<int>("robot_size", 30);
+    m_goalSize = getParam<int>("goal_size", 20);
+    m_laserScanSize = getParam<int>("laser_scan_size", 6);
+
+    m_centeredRobot = getParam<bool>("centered_robot", true);
```
Comment (philippewarren) [inline map_image_generator/src/Parameters.cpp:105]: Everything is uniform.

File: map_image_generator/src/Parameters.cpp [hunk 5]
```diff
@@ -3,62 +3,106 @@
...
-    nodeHandle.param("text_color", textColorString, string("255 255 255"));
-
-    m_wallColor = parseColorVec3b(wallColorString);
-    m_freeSpaceColor = parseColorVec3b(freeSpaceColorString);
-    m_unknownSpaceColor = parseColorVec3b(unknownSpaceColorString);
-    m_globalPathColor = parseColorScalar(globalPathColorString);
-    m_robotColor = parseColorScalar(robotColorString);
-    m_goalColor = parseColorScalar(goalColorString);
-    m_laserScanColor = parseColorScalar(laserScanColorString);
-    m_textColor = parseColorScalar(textColorString);
```
Comment (philippewarren) [inline map_image_generator/src/Parameters.cpp:55]: That part was ugly to me.
Adding a new colour required to add 3 lines.

File: map_image_generator/include/map_image_generator/Parameters.h [hunk 6]
```diff
@@ -26,7 +26,7 @@ namespace map_image_generator
         bool m_drawOccupancyGrid;
         bool m_drawGlobalPath;
         bool m_drawRobot;
-        bool m_drawGoal;
+        bool m_drawGoals;
```
Comment (philippewarren) [inline map_image_generator/include/map_image_generator/Parameters.h:29]: Multiple goals are drawn now.

File: map_image_generator/src/Parameters.cpp (hunk 3 above)
Comment (doumdi) [reply map_image_generator/src/Parameters.cpp:69]: Colour -> Color ?  (US)

**Review by doumdi — APPROVED** [review]

**Review by doumdi — APPROVED** [review]

**Pull request merged**